core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -3003,6 +3003,12 @@ private[spark] object Utils extends Logging {
     resultProps
   }
 
+  /** Create a new properties object with the same values as `props` while handling null case */
```

Comment:
I'd rather say we'd be better to fix `cloneProperties` to handle null for props instead: it just throws NPE now.

Reply:
Thanks for looking at this. I agree, I don't know why I didn't do that in the first place.